### PR TITLE
Page enhancements - add page headers to mobile reports

### PR DESCRIPTION
**Pull request by joel-jeremy**
Add mobile page headers to mobile reports page + updated Page component to have a more composable design.

### CODE REVIEW

File: packages/desktop-client/e2e/rules.test.js-snapshots/Rules-creates-a-split-transaction-rule-and-makes-sure-it-is-applied-when-creating-a-transaction-1-chromium-linux.png
Comment (MatissJanis) [inline packages/desktop-client/e2e/rules.test.js-snapshots/Rules-creates-a-split-transaction-rule-and-makes-sure-it-is-applied-when-creating-a-transaction-1-chromium-linux.png:1]: Why did the amounts in the accounts change? That seems suspicious

**Comment by MatissJanis** [comment]
Not a big fan of the "back" button change in reports (desktop). Maybe we can entirely remove the back button there? None of the other pages have it. And it doesn't really save any clicks. WDYT?

--

The mobile reports page is missing a bit of bottom margin I think. The footer overlays the last report when scrolling all the way to the bottom.

Screenshot 2024-05-10 at 19 57 33

---

Missing some bottom margin in the single report pages too:

Screenshot 2024-05-10 at 19 58 26

---

I think the settings page bottom margin should also be raised. Still no way to access the advanced settings due to footer overlaying.

Screenshot 2024-05-10 at 19 59 13

---

Other than that looks good! Code was good :+1:

**Comment by youngcw** [comment]
I don't know if this would be for a different PR, but would it be possible to make the new report button not be its own section?  Could it line up with the Reports title?
image

**Comment by joel-jeremy** [comment]
@MatissJanis @youngcw  I've removed the back button, moved the create new report button and updated the mobile pages.

**Comment by MatissJanis** [comment]
The reports page header is not using the mobile layout for the page header. What makes matters worse - when you scroll in the page - it overlays the graphs in an ugly way.

Screenshot 2024-05-14 at 19 05 43

**Comment by MatissJanis** [comment]
One more thing: looks like the reports inner pages are no longer showing any charts.

Screenshot 2024-05-14 at 19 12 32

File: packages/desktop-client/src/components/reports/reports/NetWorth.jsx [hunk 1]
```diff
@@ -150,13 +161,12 @@ export function NetWorth() {
           </Paragraph>
           <Paragraph>
             Net worth shows the balance of all accounts over time, including all
-            of your investments. Your “net worth” is considered to be the amount
-            you’d have if you sold all your assets and paid off as much debt as
-            possible. If you hover over the graph, you can also see the amount
-            of assets and debt individually.
+            of your investments. Your “net worth” is considered to be the debt
+            as possible. If you hover over the graph, you can also see the
+            amount of assets and debt individually.
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/reports/NetWorth.jsx:166]: **🔨 warning:** ‏oops? Accidentally removed some text here I think

Comment (joel-jeremy) [reply packages/desktop-client/src/components/reports/reports/NetWorth.jsx]: Oops. Didn't notice auto fix deleted them.

**Comment by joel-jeremy** [comment]
> One more thing: looks like the reports inner pages are no longer showing any charts.
Could it be because of the screen size being too small? Here's what's I'm seeing (Pixel 7 and iPhone SE screen sizes)
image
image

**Comment by MatissJanis** [comment]
Iphone 14:

image
image

The net worth actually looks Ok. Only the cash flow one is problematic on this screen size

**Comment by MatissJanis** [comment]
Cashflow graph is still tiny in iPhone 14. Perhaps we can set `minHeight` for the graphs?

**Comment by joel-jeremy** [comment]
It seems to render well on the latest push when I tested with iPhone SE which has smaller screen.

**Review by MatissJanis — APPROVED** [review]
Needed to clear cache. Works as expected now!

**Pull request merged**